### DISCUSSION TITLE
How to set fixed number of workspaces on Startup?

### DISCUSSION THREAD

**Discussion by YerongLi** [i3/i3#6180] [Q&A]
I want to set 3 workspaces on startup. And it is better to have no app started in each one of the workspaces.

Can I do this?

**Comment by okraits** [comment]
You can spawn a placeholder window on each workspace with `i3-msg open`.

**Reply by YerongLi** [reply]
Do you have specific command in the `i3/config`? For me the following command send me to a blank `Workspace 3`, this is pretty unexpected.
```
# This command will switch to workspace 1 and open a placeholder window
exec --no-startup-id "i3-msg 'workspace 1; exec i3-msg open .'"

# This command will switch to workspace 2 and open a placeholder window
exec --no-startup-id "i3-msg 'workspace 2; exec i3-msg open .'"

# This command will switch to workspace 3 and open a placeholder window
exec --no-startup-id "i3-msg 'workspace 3; exec i3-msg open .'"
```
![Screenshot from 2024-07-29 02-58-35](https://github.com/user-attachments/assets/0c1c99d1-468d-4615-860d-8930478f287b)

**Reply by okraits** [reply]
As an example, this works for me:

> exec i3-msg workspace 3; sleep 1; i3-msg open; sleep 1; i3-msg workspace 4; sleep 1; i3-msg open; sleep 1; i3-msg workspace 1

**Comment by orestisfl** [comment]
To always have workspace buttons 1 to 3 show in your bar you can use the new [workspace button protocol](https://i3wm.org/docs/i3bar-workspace-protocol.html)

**Reply by YerongLi** [reply]
So complicated, we just want workspaces... Who developed this? Why the develop don't wrap this with chatgpt?
